### ISSUE TITLE
[Navigator] pop animations hang and lag slowly in 0.14.2

### ISSUE THREAD

**Issue by jforaker** [facebook/react-native#4132]
https://www.dropbox.com/s/6ze2xllvmzrrk79/2015_11_15_16_45_56.mp4?dl=0

Ever since upgrade to 0.14.2, Navigator `pop()` animations hang and look laggy overall. As you can see in the video the view being popped "hangs" for noticeably too long, then abruptly disappears. This is not ideal and seems like a regression in terms of performance, and or is a definite bug. 

Yes it is Genymotion, yes dev mode is false, tested on brand new Nexus 5x - same thing. It was flawless before in 0.13.x
(Disregard the animation lag in the view pushes, I'll chalk that up to the emulator performance. Just pay attention to the back arrow clicks and the 'X' button clicks)

First view being pushed is:

```
sceneConfig: Navigator.SceneConfigs.FloatFromBottom
```

Second one is:
```
Navigator.SceneConfigs.FloatFromRight
```

![screen-recording-2015-11-15-16-59-33-qk4rbd8yrs](https://cloud.githubusercontent.com/assets/2042031/11171293/91d09bde-8bba-11e5-9449-7ebfd3db2c5a.gif)

**Comment by praisegeek** [comment]
It seems to affect Scene Configs with longer transitions. 
You can use Navigator.SceneConfigs.FadeAndroid for now until a fix has been made.

**Comment by brentvatne** [comment]
cc @mkonicek this is an example of where offloading animations is good ;)

**Comment by brentvatne** [comment]
Not seeing any changes to Navigator.js that landed in 0.14 that would have impacted this. Are you sure that this was the only change that you made? Are you not maybe running some code in response to navigator focus events?

**Comment by jforaker** [comment]
Can you elaborate on what you mean by "navigator focus events"?

**Comment by brentvatne** [comment]
https://facebook.github.io/react-native/docs/navigator.html#ondidfocus

**Comment by jforaker** [comment]
Definitely not, never used that before. As far as I understand, I am just doing simple pushes and pops, and passing in `sceneConfig` to mix it up the animations for the user.

**Comment by satya164** [comment]
Is this fixed in master? Cause this is there in 0.15.0-rc

The relevant issue - https://github.com/facebook/react-native/issues/3669

**Comment by brentvatne** [comment]
Thanks for additional context @satya164! Are you guys only experiencing this on Android?

**Comment by satya164** [comment]
@brentvatne I've not tested on iOS yet.

**Comment by brentvatne** [comment]
@mkonicek - are you seeing this at all internally? not sure who should be assigned

**Comment by jforaker** [comment]
Same, not tested on iOS for me.

Here is a higher res/better video showing the problem:  https://www.dropbox.com/s/6ze2xllvmzrrk79/2015_11_15_16_45_56.mp4?dl=0

Also note the crazy angled animation on the last view that occurs with:
```
sceneConfig: Navigator.SceneConfigs.VerticalUpSwipeJump
```

**Comment by 6i6arka** [comment]
+1 I can confirm this is happening on Android. Just simple push and pop, nothing else.

**Comment by kmagiera** [comment]
This issue should be fixed in 0.16 as of #3954. @jforaker, would you be able to verify that on your app and close this issue? Thanks!

**Comment by jforaker** [comment]
Installing `react-native@latest` (which was 0.16.0-rc) did not work at all for me, see https://github.com/facebook/react-native/issues/4488#issuecomment-161529113. 

`0.15.0` is the most recent version that I can actually get to run successfully without red screen errors on start. Therefore I cannot say if this issue is fixed or not.

Is there a way to install this specific "version" that I am unaware of? Installing a library at a certain pull request is not the most straightforward thing to do... 

@satya164 - have you got it to work?

**Comment by satya164** [comment]
@jforaker I cannot upgrade to RN to 0.16.0-rc due to babel issues #4327 

But if I cherry-pick b12117a03847f8ead4a202c863a0000abe6abd34 then the issue goes away.

**Comment by satya164** [comment]
Closing this since this has been fixed with b12117a
